server/src/main/java/org/elasticsearch/index/search/QueryStringQueryParser.java

```diff
@@ -264,6 +265,16 @@ private Query applyBoost(Query q, Float boost) {
             // Filters unsupported fields if a pattern is requested
             // Filters metadata fields if all fields are requested
             return resolveMappingField(context, field, 1.0f, !allFields, !multiFields, quoted ? quoteFieldSuffix : null);
+        } else if (quoted && quoteFieldSuffix != null) {
```

Comment:
When looking at how the field suffix is handled in SimpleQueryStringQuery#newPhraseQuery I saw there is a helper called  QueryParserHelper.resolveMappingFields which might be reused here. At least the new unit tests pass when using that helper. It does a little more than the code here, so I'm not sure if that would be a good idea though.

Reply:
Good call, I pushed https://github.com/elastic/elasticsearch/pull/29332/commits/0abf44c2e0dded87f69b1fdf67dc7820f13cb433

Reply:
Thanks, thats what I tried locally. I wasn't sure it works in this case though because it also seems to resolve some regex patterns around wildcards etc... and I thought maybe this isn't wanted here, but glad it can be reused.